src/server.c

```diff
@@ -4589,10 +4588,11 @@ int writeCommandsDeniedByDiskError(void) {
             return DISK_ERROR_TYPE_AOF;
         }
         /* AOF fsync error. */
-        int aof_bio_fsync_status;
-        atomicGet(server.aof_bio_fsync_status,aof_bio_fsync_status);
+        int aof_bio_fsync_status = atomic_load_explicit(&server.aof_bio_fsync_status, memory_order_relaxed);
+        atomic_thread_fence(memory_order_acquire);
```

Comment:
My suggestion was to put the fence inside the `if` block, because it's only needed if you are going to access _errno. It that way you incur no performance penalty when _status != C_ERR (not that this is necessarily important).
If you think that the code would be clearer and more maintainable applying the fence unconditionally (I can agree with this), you can just use `memory_order_acquire` directly in `atomic_load_explicit` instead of `memory_order_relaxed` and drop the `atomic_thread_fence` call.

Reply:
For clarity sake, the best would be to use memory_order_acquire directly in atomic_load_explicit.